### PR TITLE
Fe/documentation/review

### PR DESCRIPTION
**Pull request by FrancescElies**
Start #257 

For easier review I'll write the questions I have inside the commit message, this way they will be self contained and we won't have to make back references to certain lines in the files, would you be ok with that? e.g. ac85489.

### CODE REVIEW

**Comment by FrancescElies** [comment]
specially b082ea5 and e80bb8c would require a second pair of eyes.
The rest of the commits are mostly small differences in cbytes due to c-blosc version

**Comment by FrancescElies** [comment]
since it strongly depends on platform, shall I leave out those benchmark updates?
So let's say only change lines where it is obviously an old version of bcolz, stuff like cbytes or mode=w

File: doc/tutorial.rst [hunk 1]
```diff
@@ -361,16 +383,16 @@ However, you must be aware that modifying a carray is expensive::
   >>> a = np.arange(1e7)
   >>> b = bcolz.carray(a)
   >>> %timeit a[2] = 3
-  10000000 loops, best of 3: 101 ns per loop
+  10000000 loops, best of 3: 94.4 ns per loop
   >>> %timeit b[2] = 3
-  10000 loops, best of 3: 161 µs per loop  # 1600x slower than NumPy
+  1000 loops, best of 3: 274 µs per loop # 2900x slower than NumPy
 
 although modifying values in latest chunk is somewhat more cheaper::
```
Comment (FrancescAlted) [inline doc/tutorial.rst]: "somewhat more cheaper" -> "somewhat cheaper"

**Comment by FrancescAlted** [comment]
No, rather the contrary.  Keeping the benchmarks updated with more modern computers is good, IMO.  So no need to change your current commits.

**Comment by FrancescAlted** [comment]
I just had a shower and came to the realization that converting the tutorial into a Jupyter notebook can be the solution that we are looking for.  Jupyter notebooks can be rendered automatically in github since some months ago, and regenerating the output from it prier each release should be very easy.  One can even add some plots for some benchmarks too so as to make the read not that boring.  Ah, the showers :)

**Comment by FrancescElies** [comment]
That sounds an appealing idea, something similar to bquery notebook http://nbviewer.ipython.org/github/visualfabriq/bquery/blob/ipynb_bench/bquery/benchmarks/bench_groupby.ipynb ?

Going to the swimming pool, let's see if there I can get some ideas too :)

**Comment by FrancescElies** [comment]
I just found the github link https://github.com/visualfabriq/bquery/blob/ipynb_bench/bquery/benchmarks/bench_groupby.ipynb, would be this what you mean?

**Comment by FrancescElies** [comment]
It will sound familiar to you since this was strongly based on the following notebooks
Blaze - http://nbviewer.ipython.org/url/blaze.pydata.org/notebooks/timings-bcolz.ipynb
Bcolz - http://nbviewer.ipython.org/github/Blosc/movielens-bench/blob/master/querying-ep14.ipynb

**Comment by FrancescAlted** [comment]
Yes, that's exactly what I meant.  In fact, I did something along these lines for past EuroPython:

https://github.com/FrancescAlted/EuroPython2015/blob/master/0-Intro.ipynb

Initially I wanted all the docs in a Sphinx format, but now I realize that an IPython notebook is probably a much better fit for the tutorial chapters.

**Comment by FrancescElies** [comment]
from my side +1 for the notebook. Shall we close this one?

**Comment by FrancescAlted** [comment]
No, let's merge it and start work on a replacement.

**Pull request merged**